x-pack/legacy/plugins/apm/public/components/shared/charts/Timeline/Marker/ErrorMarker.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiPopover } from '@elastic/eui';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import {
+  TRACE_ID,
+  TRANSACTION_ID
+} from '../../../../../../common/elasticsearch_fieldnames';
+import { useUrlParams } from '../../../../../hooks/useUrlParams';
+import { fontSizes, px, unit } from '../../../../../style/variables';
+import { asDuration } from '../../../../../utils/formatters';
+import { ErrorDetailLink } from '../../../Links/apm/ErrorDetailLink';
+import { Legend, Shape } from '../../Legend';
+import { IWaterfallItemError } from '../../../../app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers';
+
+interface Props {
+  mark: IWaterfallItemError;
+}
+
+const Popover = styled.div`
+  max-width: ${px(280)};
+`;
+
+const Label = styled.div`
```

Comment:
As this legend style is the same that we use in our charts and Timeline "Services" legend, would be able to simply use the same component from here https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/apm/public/components/shared/charts/Legend/index.js

Reply:
I also refactored the name of this styled component to TimeLabel, as it's used to show the time (`@ 100 ms`). The Legend component isn't the best option here, as it adds a colored indicator before the text. But I'm using it to show the service name. https://github.com/cauemarcondes/kibana/blob/waterfall-erros-timeline/x-pack/legacy/plugins/apm/public/components/shared/charts/Timeline/Marker/ErrorMarker.tsx#L74

Reply:
Ah, I see.. alright, sounds good. Just want to make sure that if we were ever to restyle the legends in the Timeline header, these labels would also have that same styling.

Reply:
ah, I just saw that Dario refactored that component. I'm going to rebase my code and see if I can use it for the time as well.

Reply:
Just change to also use the Legend component @formgeist